io/src/main/scala/fs2/io/file/file.scala

```diff
@@ -227,6 +231,32 @@ package object file {
   def deleteIfExists[F[_]: Sync: ContextShift](blocker: Blocker, path: Path): F[Boolean] =
     blocker.delay(Files.deleteIfExists(path))
 
+  /**
+    * Recursively delete a directory
+    */
+  def deleteDirectoryRecursively[F[_]: Sync: ContextShift](
+      blocker: Blocker,
+      path: Path,
+      options: Set[FileVisitOption] = Set.empty
+  ): F[Unit] =
+    blocker.delay {
+      Files.walkFileTree(
+        path,
+        options.asJava,
+        Int.MaxValue,
+        new SimpleFileVisitor[Path] {
+          override def visitFile(path: Path, attrs: BasicFileAttributes): FileVisitResult = {
+            Files.deleteIfExists(path)
+            FileVisitResult.CONTINUE
+          }
+          override def postVisitDirectory(path: Path, e: IOException): FileVisitResult = {
+            Files.deleteIfExists(path)
+            FileVisitResult.CONTINUE
+          }
+        }
+      )
+    }
```

Comment:
I think this could also be done with something like:
```scala
walk(blocker, path, options)
  .compile
  .toVector
  .flatMap(_.reverse.traverse_(deleteIfExists[F](blocker, _)))
```
but thought it might have issues with very largely nested directory structures?